File: app/src/main/java/com/productivity/habits/ui/common/ColorUtils.kt
package com.productivity.habits.ui.common

import androidx.compose.ui.graphics.Color

object ColorUtils {

    fun parseHexColor(hexString: String?, defaultColor: Color = Color(0xFF10B981)): Color {
        if (hexString.isNullOrBlank()) return defaultColor
        return try {
            val cleanHex = hexString.trim().removePrefix("#")
            when (cleanHex.length) {
                6 -> {
                    val colorInt = cleanHex.toLong(16)
                    Color(0xFF000000 or colorInt)
                }
                8 -> {
                    val colorInt = cleanHex.toLong(16)
                    Color(colorInt)
                }
                else -> defaultColor
            }
        } catch (e: Exception) {
            defaultColor
        }
    }
}

File: lib/ui/common/color_utils.dart
import 'package:flutter/material.dart';

class ColorUtils {
  static Color parseHexColor(
    String? hexString, {
    Color defaultColor = const Color(0xFF10B981),
  }) {
    if (hexString == null || hexString.trim().isEmpty) {
      return defaultColor;
    }
    try {
      var cleanHex = hexString.trim();
      if (cleanHex.startsWith('#')) {
        cleanHex = cleanHex.substring(1);
      }
      if (cleanHex.length == 6) {
        final colorInt = int.parse(cleanHex, radix: 16);
        return Color(0xFF000000 | colorInt);
      } else if (cleanHex.length == 8) {
        final colorInt = int.parse(cleanHex, radix: 16);
        return Color(colorInt);
      } else {
        return defaultColor;
      }
    } catch (_) {
      return defaultColor;
    }
  }

  static Color fromHex(
    String? hexString, {
    Color defaultColor = const Color(0xFF10B981),
  }) =>
      parseHexColor(hexString, defaultColor: defaultColor);
}
